### PR TITLE
Dcmqi seg

### PR DESCRIPTION
**Pull request by jadh4v**

### CODE REVIEW

File: packages/dicom/dcmtk/write-overlapping-segmentation.cxx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    cerr << "Error: No input DICOM files specified!" << endl;
+    return EXIT_FAILURE;
+  }
+
+  // DCMQI only takes scalarImageTypes as input. Extract vector image segments as individual scalar images.
+  std::vector<ScalarImageType::Pointer> segmentations;
+
+  for(size_t idx = 0; idx < inputImage.Get()->GetNumberOfComponentsPerPixel(); ++idx)
+  {
+    auto indexSelectionFilter = itk::VectorIndexSelectionCastImageFilter<VectorImageType, ScalarImageType>::New();
```
Comment (thewtex) [inline packages/dicom/dcmtk/write-overlapping-segmentation.cxx:69]: https://itk.org/Doxygen/html/classitk_1_1VectorImageToImageAdaptor.html

File: packages/dicom/dcmtk/write-overlapping-segmentation.cxx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+#if !defined(NDEBUG) || defined(_DEBUG)
+    // Display DCMTK debug, warning, and error logs in the console
+    // For some reason, this code has no effect if it is called too early (e.g., directly after PARSE_ARGS)
+    // therefore we call it here.
+    dcmtk::log4cplus::BasicConfigurator::doConfigure();
+#endif
+
+  if(!helper::pathsExist(dicomImageFiles))
+  {
```
Comment (thewtex) [inline packages/dicom/dcmtk/write-overlapping-segmentation.cxx:100]: print an error message here

File: packages/dicom/dcmtk/write-segmentation.cxx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+#include "itkSmartPointer.h"
+#include "itkInputTextStream.h"
+
+typedef dcmqi::Helper helper;
+constexpr unsigned int Dimension = 3;
+using PixelType = short;
+using ScalarImageType = itk::Image<PixelType, Dimension>;
+
+int runPipeline(
+  itk::wasm::Pipeline & pipeline,
```
Comment (thewtex) [inline packages/dicom/dcmtk/write-segmentation.cxx:40]: remove?

File: packages/dicom/dcmtk/write-segmentation.cxx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+  itk::wasm::InputImage<ScalarImageType>& inputImage,
```
Comment (thewtex) [inline packages/dicom/dcmtk/write-segmentation.cxx:41]: const?

File: packages/dicom/dcmtk/write-segmentation.cxx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+  itk::wasm::InputTextStream& metaInfoStream,
```
Comment (thewtex) [inline packages/dicom/dcmtk/write-segmentation.cxx:42]: const?

**Comment by thewtex** [comment]
Addresses https://github.com/InsightSoftwareConsortium/ITK-Wasm/issues/991

File: packages/dicom/dcmtk/write-segmentation.cxx (hunk 5 above)
Comment (jadh4v) [reply packages/dicom/dcmtk/write-segmentation.cxx:41]: Might need to add a .Get() const function to InputTextStream class. I will run an experiment to see if it builds fine.

File: packages/core/typescript/itk-wasm/src/bindgen/typescript/typescript-bindings.js [hunk 6]
```diff
@@ -36,6 +36,7 @@ function typescriptBindings(
   options,
   forNode = false
 ) {
+  console.log('wasmBinaries: ', wasmBinaries);
```
Comment (thewtex) [inline packages/core/typescript/itk-wasm/src/bindgen/typescript/typescript-bindings.js:39]: debug code

File: packages/dicom/CMakeLists.txt [hunk 7]
```diff
@@ -5,5 +5,5 @@ set(CMAKE_CXX_STANDARD 20)
 
 enable_testing()
 
-add_subdirectory(gdcm)
+#add_subdirectory(gdcm)
```
Comment (thewtex) [inline packages/dicom/CMakeLists.txt:8]: uncomment

File: packages/dicom/dcmtk/CMakeLists.txt [hunk 8]
```diff
@@ -1,10 +1,40 @@
...
   )
 include(${ITK_USE_FILE})
 
+if(POLICY CMP0077)
+  set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
+endif()
+
+include(FetchContent)
+# DCMQI
+set(DCMQI_GIT_REPOSITORY "https://github.com/jadh4v/dcmqi.git")
```
Comment (thewtex) [inline packages/dicom/dcmtk/CMakeLists.txt:16]: update to upstream if possible

File: packages/dicom/package.json [hunk 9]
```diff
@@ -28,7 +28,7 @@
     "build:python:versionSync": "itk-wasm pnpm-script build:python:versionSync",
     "publish:python": "itk-wasm pnpm-script publish:python",
     "test": "pnpm test:data:download && pnpm build:gen:python && pnpm test:python",
-    "test:data:download": "dam download test/data test/data.tar.gz bafybeicskxufnvuem6342pkfwgeo3siiozgzmfo5f34woge6aptuzuwzzu https://github.com/InsightSoftwareConsortium/ITK-Wasm/releases/download/itk-wasm-v1.0.0-b.119/dicom-test-data.tar.gz https://w3s.link/ipfs/bafybeiby67winzvozowf4moqthwunuxxscssitnb6wahxv4ugvfxhu2vki/data.tar.gz",
+    "test:data:download": "dam download test/data test/data.tar.gz bafybeifswmu56lqciufqydkgjhcjbamivextuqylpjdh672ivnrmispgfy https://data.kitware.com/api/v1/file/66be71dcf87a980650f422a1/download",
```
Comment (thewtex) [inline packages/dicom/package.json:31]: after merge, please also upload to the latest GitHub published Release and create a new PR adding the URL

**Review by thewtex — CHANGES_REQUESTED** [review]

File: packages/dicom/dcmtk/write-segmentation.cxx (hunk 5 above)
Comment (jadh4v) [reply packages/dicom/dcmtk/write-segmentation.cxx]: I think this one is a little more complicated than I thought. We should leave it as non-const for now.
The `std::istream` object possibly self-modifies itself when its being read which is why there's no const way to convert it into an `std::string`.

File: packages/dicom/dcmtk/CMakeLists.txt (hunk 8 above)
Comment (jadh4v) [reply packages/dicom/dcmtk/CMakeLists.txt:16]: Will update in a separate PR after dcmqi branch is merged, see here:
https://github.com/QIICR/dcmqi/pull/507

File: packages/dicom/package.json (hunk 9 above)
Comment (jadh4v) [reply packages/dicom/package.json]: tracking in issue #1202

**Pull request merged**